tests/director/tests/runtime_contexts_api_test.go

```diff
@@ -163,275 +161,264 @@ func TestDeleteRuntimeContext(t *testing.T) {
 }
 
 func TestRuntimeContextSubscriptionFlows(stdT *testing.T) {
+	testCases := []struct {
+		FirstSubscriptionFlow       string
+		SecondSubscriptionFlow      string
+		SubscriptionProviderID      string
+		SubscribedToProviderAppName string // this is provider app name which we subscribe for
+		ExpectedProviderAppName     string // this is the value of runtimeType label of the runtime from which we create runtime context. For the tests where we are subscribing for an indirect dependency this should be the app name of the direct dependency.
+	}{
+		{
+			FirstSubscriptionFlow:       subscription.StandardFlow,
+			SecondSubscriptionFlow:      subscription.StandardFlow,
+			SubscriptionProviderID:      conf.SubscriptionConfig.SelfRegDistinguishLabelValue,
+			SubscribedToProviderAppName: conf.SubscriptionProviderAppNameValue,
+			ExpectedProviderAppName:     conf.SubscriptionProviderAppNameValue,
+		},
+		{
+			FirstSubscriptionFlow:       subscription.IndirectDependencyFlow,
+			SecondSubscriptionFlow:      subscription.DirectDependencyFlow,
```

Comment:
What's the idea here - to have two subscriptions - direct and indirect and validate with both of them that the subscription label is properly increased/decreased?

Reply:
The values are used to specify to which participant should we subscribe. The participants and their relationships are Indirect ->depends on  Direct ->depends on CMP. The test makes two subscriptions - the first one will be to the SaaS app Indirect(whisch should result to callback to Direct and then callback to CMP) and we will create subscriptions label with the subscription ID. Then a subscription to the Saas app direct will be done and a callback to CMP will be sent. Then we will add the subscription ID to the label and the label will now contain both of the subscriptions.